jetty-core/jetty-http2/jetty-http2-server/src/main/java/org/eclipse/jetty/http2/server/internal/HTTP2ServerConnection.java

```diff
@@ -74,6 +76,8 @@ public HTTP2ServerConnection(Connector connector, EndPoint endPoint, HttpConfigu
         this.listener = listener;
         this.httpConfig = httpConfig;
         this.id = StringUtil.randomAlphaNumeric(16);
+        localSocketAddress = httpConfig.getLocalAddress() != null ? httpConfig.getLocalAddress() : endPoint.getLocalSocketAddress();
+        remoteSocketAddress = endPoint.getRemoteSocketAddress();
```

Comment:
Seems like we can have a null localSocketAddress and remoteSocketAddress this way, is that wise?

Reply:
Always could.   If there is no address, there is no address.